### PR TITLE
Bump MSRV to 1.40

### PR DESCRIPTION
**Pull request by eduardosm**
This how things would look if we bump the MSRV to 1.40.

@psychon Feel free to accept or reject it.

### CODE REVIEW

File: examples/tutorial.rs [hunk 1]
```diff
@@ -221,8 +221,7 @@ fn example1() -> Result<(), Box<dyn Error>> {
     let start = Instant::now();
     let cookies = names
         .iter()
-        .map(|name| conn.intern_atom(false, name.as_bytes()))
-        .collect::<Vec<_>>();
+        .map(|name| conn.intern_atom(false, name.as_bytes()));
     for (i, atom) in cookies.into_iter().enumerate() {
```
Comment (psychon) [inline examples/tutorial.rs:225]: Well... no. This defeats the point of this example (which is to send all the `InternAtom` requests before starting to wait for their replies). Now, we are back to "send one request and wait for its reply". Hopefully this difference shows up when running the example and I am not just imagining it...

**Review by psychon — COMMENTED** [review]
The first commit is quite convincing. Lots of hacks going away, so :+1: 
Only nit is in the last commit since I think we really want to "pay" that one memory allocation if it saves 99% of round-trips.

Thanks for doing this. "Show me the code/diff" is surprising convincing :-)

File: examples/tutorial.rs (hunk 1 above)
Comment (eduardosm) [reply examples/tutorial.rs:225]: Yep, you're right, I didn't realize that the `collect` was to actually execute the closure passed to `map`.

**Review by psychon — APPROVED** [review]

**Comment by psychon** [comment]
The Travis build took forever to finish (and of course still fails to report its status to GitHub) :-(

**Pull request merged**